packages/experimental-utils/src/ts-eslint/Rule.ts

```diff
@@ -209,13 +209,13 @@ interface RuleContext<
    * the root of the AST and continuing through the direct parent of the current node.
    * This array does not include the currently-traversed node itself.
    */
-  getAncestors(): TSESTree.Node[];
+  getAncestors(): readonly TSESTree.Node[];
 
   /**
    * Returns a list of variables declared by the given node.
    * This information can be used to track references to variables.
    */
-  getDeclaredVariables(node: TSESTree.Node): Scope.Variable[];
+  getDeclaredVariables(node: TSESTree.Node): readonly Scope.Variable[];
```

Comment:
this change is fine - `getDeclaredVariables` comes from the scope manager (i.e. via the parser):
https://github.com/eslint/eslint/blob/d867f8100737bb82742debee2b5dc853c5f07c91/lib/linter/linter.js#L875

So it's dependent on the scope manager's implementation.
I.e. in our scope manager it should be a readonly array because we don't return a new array:
https://github.com/typescript-eslint/typescript-eslint/blob/1c1b572c3000d72cfe665b7afbada0ec415e7855/packages/scope-manager/src/ScopeManager.ts#L92-L94